README.md

```diff
@@ -273,9 +282,150 @@ dependencies:
+
+### Using Haskell Language Server with VS Code
+
+Install from
+[the VSCode marketplace](https://marketplace.visualstudio.com/items?itemName=alanz.vscode-hie-server), or manually from the repository [vscode-hie-server](https://github.com/alanz/vscode-hie-server).
+
+Choose `haskell-language-server` in the extension setting `languageServerHaskell.hieVariant`.
+
+### Using Haskell Language Server with Sublime Text
+
+- Make sure haskell-language-server and haskell-language-server-wrapper are installed (see above) and they are in the global `$PATH`.
+- Install [LSP](https://packagecontrol.io/packages/LSP) using [Package Control](https://packagecontrol.io/)
+- From Sublime Text, press Command+Shift+P and search for Preferences: LSP Settings
+- Paste in these settings. Make sure to change the command path to your `hie`
+
+```json
+{
+"clients": {
+  "haskell-ide-engine": {
+    "command": ["haskell-language-server-wrapper", "--lsp"],
+    "scopes": ["source.haskell"],
+    "syntaxes": ["Packages/Haskell/Haskell.sublime-syntax"],
+    "languageId": "haskell",
+  },
+},
+}
+```
+
+Now open a Haskell project with Sublime Text. You should have these features available to you:
+
+1. Errors are underlined in red
+2. LSP: Show Diagnostics will show a list of hints and errors
+3. LSP: Format Document will prettify the file
+
+### Using Haskell Language Server with Vim or Neovim
+
+As above, make sure haskell-language-server and haskell-language-server-wrapper are installed.
+Then you can use [Coc](https://github.com/neoclide/coc.nvim), [LanguageClient-neovim](https://github.com/autozimu/LanguageClient-neovim)
+or any other vim Language server protocol client.
+Coc is recommend since it is the only complete LSP implementation for Vim and Neovim and offers snippets and floating documentation out of the box.
+
+#### Coc
+
+Follow Coc's [installation instructions](https://github.com/neoclide/coc.nvim),
+Then issue `:CocConfig` and add the following to your Coc config file.
+
+```jsonc
+"languageserver": {
+  "haskell": {
+    "command": "haskell-language-server-wrapper",
+    "args": ["--lsp"],
+    "rootPatterns": [
+      "*.cabal",
+      "stack.yaml",
+      "cabal.project",
+      "package.yaml"
+    ],
+    "filetypes": [
+      "hs",
+      "lhs",
+      "haskell"
+    ],
+    "initializationOptions": {
+      "languageServerHaskell": {
+      }
+    }
+  }
+}
+```
+
+#### LanguageClient-neovim
+
+##### vim-plug
+
+If you use [vim-plug](https://github.com/junegunn/vim-plug), then you can do this by e.g.,
+including the following line in the Plug section of your `init.vim` or `~/.vimrc`:
+
+```text
+Plug 'autozimu/LanguageClient-neovim', {
+    \ 'branch': 'next',
+    \ 'do': './install.sh'
```

Comment:
@anrock not sure if this configuration is working anymore, even with hie, there is no install.sh

Reply:
This is how I install `LanguageClient-neovim` by using `vim-plug`:
```
Plug 'autozimu/LanguageClient-neovim', {
    \ 'branch': 'next',
    \ 'do': 'bash install.sh',
    \ }
```
so it should work. But it seems kind of redundant to have installations instructions for it copied here. Wouldn't a link ([this one?](https://github.com/autozimu/LanguageClient-neovim/blob/next/INSTALL.md#vim-plug-user)) be better?

Reply:
@jneira I've migrated to CoC, so no idea.

Reply:
well i think we are copying instructions for other editors too, feel free to open a new pr replacing them with the link (or add it)